### PR TITLE
Make target to download binaries dependencies

### PR DESCRIPTION
**Pull request by luizbafilho**
I mostly brought the scripts from the wks repo.

### CODE REVIEW

File: tools/download-deps.sh [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    run_stoml() {
+        local property="${1}"
+        "${bin_dir}"/stoml "${dependencies_toml}" "${tool}"."${property}"
+    }
+
+    local url_and_path
+    local fetch
+    local binarypath
+	local special_tarpath
+	local special_tarpath_url
```
Comment (palemtnrider) [inline tools/download-deps.sh:68]: Nit formatting

File: tools/download-deps.sh [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    binarypath=$(instantiate_url "$(run_stoml binarypath)")
+	special_tarpath=$(instantiate_url "$(run_stoml special_tarpath)")
+	special_tarpath_url=(${special_tarpath//;/ }) # split out special paths which contain <url>;<path in tarball>
+    local tarpath
+    tarpath=$(instantiate_url "$(run_stoml tarpath)")
+    echo $tarpath
```
Comment (palemtnrider) [inline tools/download-deps.sh:74]: Would it be helpful to print the other paths/urls?  Maybe print them inside the if statements below?

File: tools/download-deps.sh [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        url_and_path="${binarypath}"
+        fetch=do_curl_binary
+    elif check_url "${special_tarpath_url}"; then
+        url_and_path="${special_tarpath}"
+        fetch=do_curl_tarball_with_path
+    elif check_url "${tarpath}"; then
+        url_and_path="${tarpath}"
+        fetch=do_curl_tarball
+    else
+        echo "No valid path for tool:" "${tool}"
```
Comment (palemtnrider) [inline tools/download-deps.sh:85]: Seems like we should exit here instead of trying to continue?

File: tools/download-deps.sh [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    fi
+    "${fetch}" "${tool}" "${url_and_path}" "${bin_dir}"
+}
+
+# Don't use $RELEASE_GOOS here, should be whatever is running the script.
+STOML_URL="https://github.com/freshautomations/stoml/releases/download/v0.4.0/stoml_$(goos)_amd64"
+do_curl_binary "stoml" "${STOML_URL}" "${BIN_DIR}"
+tools=(flux)
```
Comment (palemtnrider) [inline tools/download-deps.sh:93]: Is this temporary for testing just the download of flux? I was thinking the list of tools to download would come directly from the dependencies.toml file. i.e., everything listed in there is downloaded.

File: tools/functions.sh [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+        echo "386"
+        ;;
+    *)
+        error "uknown arch: ${arch}"
+        ;;
+    esac
+}
+
+mktempdir() {
+    mktemp -d 2>/dev/null || mktemp -d -t 'wk-quickstart'
```
Comment (palemtnrider) [inline tools/functions.sh:78]: Please replace wk-quickstart with something related to wego

File: tools/functions.sh [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+    local url="${2}"
+
+    log "Downloading ${url}"
+    curl --progress-bar -fLo "${path}" "${url}"
+}
+
+do_curl_binary() {
+    local cmd="${1}"
+    local url="${2}"
+    local default_path="${HOME}/.wks/bin"
```
Comment (palemtnrider) [inline tools/functions.sh:92]: replace .wks with wego (and in the other places below)

**Review by palemtnrider — CHANGES_REQUESTED** [review]
Great to have the binaries downloaded dynamically.  I made a few minor comments.  Looks like the functions file came from WKP - I'd suggest removing the functions we don't need.

**Review by J-Thompson12 — APPROVED** [review]
LGTM

**Pull request merged**